pkg/controller/cloud/node_controller.go

```diff
@@ -55,27 +55,27 @@ var labelReconcileInfo = []struct {
 	ensureSecondaryExists bool
 }{
 	{
-		// Reconcile the beta and the GA zone label using the beta label as
+		// Reconcile the beta and the GA zone label using the GA label as
 		// the source of truth
-		// TODO: switch the primary key to GA labels in v1.21
```

Comment:
Assuming this was a mistake and should have been "remove secondary key in v1.21". The primary key can be switched at any point since we can assume any >=v1.17 cluster will have both keys.

Reply:
in what release did the kubelet start adding the GA labels?

Reply:
v1.17

Reply:
Because kubelets can skew 2 versions older than the control plane/controllers, a 1.16 kubelet is possible until at least 1.19. The original comment here was on the conservative side, but 1.18 is not safe.